### PR TITLE
test: rewrite scalar expression filtering tests with eval ground truth and bug regression

### PR DESCRIPTION
**Pull request by yanliang567**
## Summary
- Rewrite `test_milvus_client_scalar_filtering.py` (2156→1567 lines) with eval-based ground truth, deterministic boundary values, and comprehensive operator/type coverage
- Add `NullValue` sentinel class in `common_func.py` for SQL NULL 3VL semantics (reusable across test files)
- 4 focused test classes replacing 1 monolithic class: Correctness, IndexConsistency, CornerCase (bug regression), JSON

## Why
The original test had fundamental gaps discovered while analyzing real bugs:
- **No boundary values** (INT64_MAX/MIN, 2^53) → missed #48440 (overflow)
- **No ground truth** (only self-consistency) → couldn't detect wrong results from all code paths
- **Hand-written validators** with incorrect NULL handling → false passes
- **No NOT+AND+OR compound coverage** → missed #48441 (3VL bitmap)
- **No bool literal / mixed-type IN tests** → missed #48443, #48442

## What's covered now

**Type coverage**: INT8/16/32/64, FLOAT, DOUBLE, BOOL, VARCHAR, JSON, ARRAY(INT8/16/32/64/FLOAT/DOUBLE/BOOL/VARCHAR)

**Operator coverage**: `==, !=, >, <, >=, <=, IN, NOT IN, LIKE (12 patterns), IS NULL, IS NOT NULL, +, -, *, /, %, **, array_contains, array_contains_all, array_contains_any, array_length, json path, json_contains`

**Compound expressions**: AND, OR, NOT, &&, ||, NOT+AND, NOT+OR, triple nesting, DeMorgan, double NOT, 3-way AND/OR, cross-field, range patterns

**Index consistency**: INVERTED, BITMAP, STL_SORT, TRIE — verified with both cross-index comparison AND ground truth

**Bug regression (L0)**: #48440 overflow (+,-,*), #48441 3VL NOT+nullable (incl. all-NULL segment), #48442 JSON mixed-type IN precision, #48443 bool literal parser

## Test results (master-20260318-3ec434d)
```
37 tests: 32 passed, 5 failed (all real Milvus bugs)
L0+L1 (19 tests, -n 4): 37 seconds
```

| Failed test | Issue |
|---|---|
| test_int64_overflow_addition | #48440 |
| test_int64_overflow_subtraction | #48440 |
| test_3vl_not_all_null_segment | #48441 |
| test_bool_literal_in_logical_expr | #48443 |
| test_json_mixed_type_in_precision | #48442 |

## Tag distribution
- **L0 (7)**: Bug regression — every PR must run
- **L1 (12)**: Core type correctness + compound + index consistency
- **L2 (18)**: Extended types + all arrays + JSON path/contains

issue: #48048

## Test plan
- [x] L0+L1 pass (minus known Milvus bugs)
- [x] L2 pass
- [x] Eval engine verified: SQL NULL 3VL, NOT propagation, NOT IN, IS NULL/IS NOT NULL
- [x] No false passes: parser errors → pytest.fail, skip rate tracking

🤖 Generated with [Claude Code](https://claude.com/claude-code)

### CODE REVIEW

File: tests/python_client/common/common_func.py [hunk 1]
```diff
@@ -49,6 +49,45 @@
...
+    def __rpow__(self, other): return self
+    def __neg__(self): return self
+    def __pos__(self): return self
+    def __abs__(self): return self
+    def __hash__(self): return id(self)
+    def __bool__(self): return False
+    def __repr__(self): return "NULL"
+
+
+SQL_NULL = NullValue()
```
Comment (wangting0128) [inline tests/python_client/common/common_func.py:88]: This code appears to be unused?

**Review by wangting0128 — APPROVED** [review]
but all seems good

/lgtm
